### ISSUE TITLE
Use apps/v1 instead of extensions/v1beta1 (required for k8s 1.16)

### ISSUE THREAD

**Issue by consideRatio** [projectcalico/calico#2915]
## Issue
As part of installing calico on a kubernetes-in-docker (kind) cluster with version 1.16.1, I ended up with an error of about the use of a now removed `apiVersion: extensions/v1beta1` for a `DaemonSet`. The change required is to use `apps/v1` instead as apiVersion.

I failed to locate where I would need to make the change, so I've not submitted a PR to fix it. I'd look for the use of extensions/v1beta1 all over and not only in these files.

## Future CI testing of these matters
[helm-kubeval](https://github.com/instrumenta/helm-kubeval) or [kubeval](https://github.com/instrumenta/kubeval) standalone can help you test for the validity of various k8s manifests on various k8s versions.

## Error
```
kubectl apply -f https://docs.projectcalico.org/v3.10/getting-started/kubernetes/installation/hosted/etcd.yaml

error: unable to recognize "https://docs.projectcalico.org/v3.10/getting-started/kubernetes/installation/hosted/etcd.yaml": no matches for kind "DaemonSet" in version "extensions/v1beta1"
```

### Valid k8s 1.16 yaml (calico.yaml)
```yaml
# from https://docs.projectcalico.org/v3.10/getting-started/kubernetes/installation/hosted/calico.yaml
# ...
# Source: calico/templates/calico-node.yaml
# This manifest installs the calico-node container, as well
# as the CNI plugins and network config on
# each master and worker node in a Kubernetes cluster.
kind: DaemonSet
apiVersion: apps/v1
# ...
```

### Invalid k8s 1.16 yaml (etcd.yaml)
```yaml
# from https://docs.projectcalico.org/v3.10/getting-started/kubernetes/installation/hosted/etcd.yaml
# ...
# This manifest installs the Calico etcd on the kubeadm master.  This uses a DaemonSet
# to force it to run on the master even when the master isn't schedulable, and uses
# nodeSelector to ensure it only runs on the master.
apiVersion: extensions/v1beta1
kind: DaemonSet
# ...
```

**Comment by caseydavenport** [comment]
@consideRatio we fixed this in Calico v3.8. Please see this similar issue here: https://github.com/projectcalico/calico/issues/2814

**Comment by consideRatio** [comment]
https://docs.projectcalico.org/v3.10/getting-started/kubernetes/installation/hosted/etcd.yaml

This specific resource tagged 3.10 still contains old stuff, the same goes for 3.9.

**Comment by caseydavenport** [comment]
Interesting. That manifest should just be removed - I don't think it's part of out installation process and hasn't been for a long time.

**Comment by consideRatio** [comment]
It is referenced here in the v3.9 documentation: https://docs.projectcalico.org/v3.9/getting-started/kubernetes/installation/calico#installing-with-the-etcd-datastore

**Comment by caseydavenport** [comment]
The `etcd.yaml` file in question is something that we used to include a long time ago to install a simple etcd cluster for use in demo systems. It was really just meant for our quickstart guide. We removed it a while back since it's not really production ready and we switched our quickstart guide to use the Kubernetes API data store instead of etcd. 

We recommend installing etcd following the upstream etcd instructions.

Also worth noting that we have a mode of operation that doesn't need a dedicated etcd cluster at all (using the Kubernetes API / CRDs).
